worker/upgradesteps/worker.go

```diff
@@ -462,23 +454,16 @@ var getUpgradeRetryStrategy = func() utils.AttemptStrategy {
 	}
 }
 
-// jobsToTargets determines the upgrade targets corresponding to the
-// jobs assigned to an agent. This determines the upgrade steps
+// upgradeTargets determines the upgrade targets corresponding to the
+// role of an agent. This determines the upgrade steps
 // which will run during an upgrade.
-func jobsToTargets(jobs []multiwatcher.MachineJob, isMaster bool) (targets []upgrades.Target) {
-	if jobs == nil {
-		return
-	}
-	for _, job := range jobs {
-		switch job {
-		case multiwatcher.JobManageModel:
-			targets = append(targets, upgrades.Controller)
-			if isMaster {
-				targets = append(targets, upgrades.DatabaseMaster)
-			}
-		case multiwatcher.JobHostUnits:
-			targets = append(targets, upgrades.HostMachine)
+func upgradeTargets(isController, isMaster bool) []upgrades.Target {
+	var targets []upgrades.Target
+	if isController {
+		targets = append(targets, upgrades.Controller)
+		if isMaster {
+			targets = append(targets, upgrades.DatabaseMaster)
 		}
 	}
-	return
+	return append(targets, upgrades.HostMachine)
```

Comment:
do we wanna add `upgrades.HostMachine` for the `controller`?

Reply:
Yes, all machines can host units